it/test/uk/gov/hmrc/tradergoodsprofilesrouter/CreateRecordIntegrationSpec.scala

```diff
@@ -0,0 +1,709 @@
+                 | {
+                 |    "invalid": "error"
+                 |  }
+                 |""".stripMargin)
+          )
+
+          val response = wsClient
+            .url(fullUrl(s"/records/"))
+            .withHttpHeaders(("Content-Type", "application/json"), ("X-Client-Id", "tss"))
+            .post(createRecordRequestData)
+            .futureValue
+
+          response.status shouldBe BAD_REQUEST
+          response.json   shouldBe Json.obj(
+            "correlationId" -> correlationId,
+            "code"          -> "UNEXPECTED_ERROR",
+            "message"       -> "Unexpected Error"
+          )
+
+          verifyThatDownstreamApiWasCalled()
+        }
+      }
+      "invalid, specifically" - {
+        "missing required header" in {
+          val response = wsClient
+            .url(fullUrl(s"/records/"))
+            .withHttpHeaders(("Content-Type", "application/json"))
+            .post(createRecordRequestData)
+            .futureValue
+
+          response.status shouldBe BAD_REQUEST
+          response.json   shouldBe Json.obj(
+            "correlationId" -> correlationId,
+            "code"          -> "BAD_REQUEST",
+            "message"       -> "Missing mandatory header X-Client-Id"
+          )
+
+          verifyThatDownstreamApiWasNotCalled()
+        }
+        "missing required request field" in {
+          val response = wsClient
+            .url(fullUrl(s"/records/"))
+            .withHttpHeaders(("Content-Type", "application/json"), ("X-Client-Id", "tss"))
+            .post(invalidRequestData)
+            .futureValue
+
+          response.status shouldBe BAD_REQUEST
+          response.json   shouldBe Json.obj(
+            "correlationId" -> correlationId,
+            "code"          -> "BAD_REQUEST",
+            "message"       -> "Bad Request",
+            "errors" -> Json.arr(
+              Json.obj(
+                "code" -> "006",
+                "message" -> "006 - Mandatory field eori was missing from body"
+              )
+            )
+          )
+
+          verifyThatDownstreamApiWasNotCalled()
+        }
+      }
+    }
+  }
+
+  private def stubForEis(httpStatus: Int, requestBody: String, responseBody: Option[String] = None) = stubFor(
+    post(urlEqualTo(s"$connectorPath"))
+      .withRequestBody(equalToJson(requestBody))
+      .withHeader("Content-Type", equalTo("application/json"))
+      .withHeader("X-Forwarded-Host", equalTo("MDTP"))
+      .withHeader("X-Correlation-ID", equalTo(correlationId))
+      .withHeader("Date", equalTo(timestamp))
+      .withHeader("Accept", equalTo("application/json"))
+      .withHeader("Authorization", equalTo("bearerToken"))
+      .withHeader("X-Client-Id", equalTo("tss"))
+      .willReturn(
+        aResponse()
+          .withHeader("Content-Type", "application/json")
+          .withStatus(httpStatus)
+          .withBody(responseBody.getOrElse(null))
+      )
+  )
+
+  val createRecordResponseData: JsValue =
```

Comment:
If you make these vals `lazy` vals the warning about a forward reference will go away.

`lazy val createRecordResponseData: JsValue`